mycelo/loadbot/bot.go

```diff
@@ -40,42 +41,71 @@ func Start(ctx context.Context, cfg *Config) error {
 		return cfg.Accounts[idx].Address, cfg.Amount
 	}
 
-	for _, acc := range cfg.Accounts {
+	// developer accounts / TPS = duration in seconds.
+	// Need the fudger factor to get up a consistent TPS at the target.
+	delay := time.Duration(int(float64(len(cfg.Accounts)*1000/cfg.TransactionsPerSecond)*0.95)) * time.Millisecond
+	startDelay := delay / time.Duration(len(cfg.Accounts))
+
+	for i, acc := range cfg.Accounts {
+		// Spread out client load across different diallers
+		client := cfg.Clients[i%len(cfg.Clients)]
+
+		err := waitFor(ctx, startDelay)
```

Comment:
i would have this delay inside the worker go routine... looks cleaner

Also, if the first go routine ends with error, you should be able to notice it right away... but you only check that on `group.Wait()`, so you only check that after doing all the delays for every routine... (i'm not being 100% true here, since waitFor will notice it becaue of the `ctx` argument.. i think... but still shady logic)

Reply:
I tried to implement this, but it significantly reduced the stability of the actual TPS to move it into the load bot function. Doing a proper start time might enable the go routine to be more accurate (`wait until`)